packages/abstractions/src/requestInformation.ts

```diff
@@ -84,18 +84,12 @@ export class RequestInformation {
   public addRequestHeaders(source: Record<string, string[]> | undefined) {
```

Comment:
request options to use this new headers class, this way we could simply pass it here, and use the add all method you've already implemented.

Reply:
I have created a new issue: https://github.com/microsoft/kiota-typescript/issues/948, since it will require both the generator and the abstraction to be updated in tandem.